jest/jest.config.shared.js

```diff
@@ -0,0 +1,14 @@
+/** @type {import('@jest/types').Config.InitialOptions} */
+module.exports = {
+  testEnvironment: "node",
+  testMatch: ["<rootDir>/**/*-test.[jt]s?(x)"],
+  setupFiles: ["<rootDir>/__tests__/setup.ts"],
+  transform: {
+    "\\.[jt]sx?$": require.resolve("./transform"),
+  },
+  watchPlugins: [
+    require.resolve("jest-watch-select-projects"),
+    require.resolve("jest-watch-typeahead/filename"),
+    require.resolve("jest-watch-typeahead/testname"),
+  ],
+};
```

Comment:
Love this!  I'd like to follow suit with the jest configs in react-router (hopefully making our eventual repo merge easier as well).

Beyond this - I'd also vote for investigating this type of "extendable monorepo config" with TS/eslint/rollup as well